pgcli/packages/function_metadata.py

```diff
@@ -30,122 +30,31 @@ def __ne__(self, other):
-
-        Field/column lists are used in function signatures and table
-        definitions. This function parses a flattened list of sqlparse tokens
-        and yields one metadata argument per argument / column.
-    """
-
-    # postgres function argument list syntax:
-    #   " ( [ [ argmode ] [ argname ] argtype
-    #               [ { DEFAULT | = } default_expr ] [, ...] ] )"
-
-    mode_names = set(('IN', 'OUT', 'INOUT', 'VARIADIC'))
-    parse_state = 'type'
-    parens = 0
-    field = TypedFieldMetadata()
-
-    for tok in tokens:
-        if tok.ttype in Whitespace or tok.ttype in Comment:
-            continue
-        elif tok.ttype in Punctuation:
-            if parens == 0 and tok.value == ',':
-                # End of the current field specification
-                if field.type:
-                    yield field
-                # Initialize metadata holder for the next field
-                field, parse_state = TypedFieldMetadata(), 'type'
-            elif parens == 0 and tok.value == '=':
-                parse_state = 'default'
-            else:
-                field[parse_state].append(tok)
-                if tok.value == '(':
-                    parens += 1
-                elif tok.value == ')':
-                    parens -= 1
-        elif parens == 0:
-            if tok.ttype in Keyword:
-                if not field.name and tok.value.upper() in mode_names:
-                    # No other keywords allowed before arg name
-                    field.mode = tok.value.upper()
-                elif tok.value.upper() == 'DEFAULT':
-                    parse_state = 'default'
-                else:
-                    parse_state = 'unknown'
-            elif tok.ttype == Name and not field.name:
-                # note that `ttype in Name` would also match Name.Builtin
-                field.name = tok.value
-            else:
-                field[parse_state].append(tok)
-        else:
-            field[parse_state].append(tok)
-
-    # Final argument won't be followed by a comma, so make sure it gets yielded
-    if field.type:
-        yield field
-
-
-def fields(sql, mode_filter=('IN', 'OUT', 'INOUT', 'VARIADIC')):
-    """Yields ColumnMetaData namedtuples from a table declaration"""
-
-    if not sql:
-        return
-
-    # sql is something like "x int, y text, ..."
-    tokens = sqlparse.parse(sql)[0].flatten()
-    for f in parse_typed_field_list(tokens):
-        if f.name and (not mode_filter or f.mode in mode_filter):
-            yield ColumnMetadata(f.name, None, [])
+        elif not self.arg_modes:
+            # For functions  without output parameters, the function name
+            # is used as the name of the output column.
+            # E.g. 'SELECT unnest FROM unnest(...);'
+            return [ColumnMetadata(self.func_name, self.return_type, [])]
+
+        return [ColumnMetadata(name, type, [])
+            for name, type, mode in zip(
```

Comment:
argh I'm sorry I forgot that `type` is a builtin function in python. Can you use `typ` or `type_` instead? My fault